airflow/dag_processing/manager.py

```diff
@@ -829,22 +830,27 @@ def _render_log_filename(self, dag_file: DagFileInfo) -> str:
     def _get_logger_for_dag_file(self, dag_file: DagFileInfo):
         log_filename = self._render_log_filename(dag_file)
         log_file = init_log_file(log_filename)
-        underlying_logger = structlog.BytesLogger(log_file.open("ab"))
+        filehandle = log_file.open("ab")
+        underlying_logger = structlog.BytesLogger(filehandle)
         processors = logging_processors(enable_pretty_log=False)[0]
-        return structlog.wrap_logger(underlying_logger, processors=processors, logger_name="processor").bind()
+        return structlog.wrap_logger(
+            underlying_logger, processors=processors, logger_name="processor"
+        ).bind(), filehandle
 
     def _create_process(self, dag_file: DagFileInfo) -> DagFileProcessorProcess:
         id = uuid7()
 
         callback_to_execute_for_file = self._callback_to_execute.pop(dag_file, [])
+        logger, filehandle = self._get_logger_for_dag_file(dag_file)
 
         return DagFileProcessorProcess.start(
             id=id,
             path=dag_file.absolute_path,
             bundle_path=cast(Path, dag_file.bundle_path),
             callbacks=callback_to_execute_for_file,
             selector=self.selector,
-            logger=self._get_logger_for_dag_file(dag_file),
+            logger=logger,
+            filehandle=filehandle,
```

Comment:
Overall looks good. I think we should rename this though. Maybe `logger_filehandle`?

Reply:
Thanks, updated.